service/src/main/java/uk/gov/hmcts/reform/fpl/service/additionalapplications/UploadAdditionalApplicationsService.java

```diff
@@ -56,7 +69,26 @@ public AdditionalApplicationsBundle buildAdditionalApplicationsBundle(CaseData c
             .build();
     }
 
-    private C2DocumentBundle buildC2DocumentBundle(CaseData caseData, String uploadedBy, LocalDateTime uploadedTime) {
+    private String getSelectedApplicantName(Object applicantsList, String otherApplicant) {
+        DynamicList dynamicList = mapper.convertValue(applicantsList, DynamicList.class);
```

Comment:
Dynamic list service should be able to give you this functionality

Reply:
although looking at what you are doing the you can probs just declare the dynamic list type in case data